content/posts/2022-11-09-iOS16-lock-screen-dynamic-time/index.md

```diff
@@ -0,0 +1,185 @@
+category: React Native
+tags:
+  - React Native CLI
+  - React Native
+  - Expo
+  - Reanimated
+  - dayjs
+keywords: react native, expo, react native cli, reanimated, UI, UX, dayjs, clock
+author: Saad
+---
+
+import YoutubeVideo from "../../../src/components/shared/YoutubeVideo/YoutubeVideo";
+import Snack from '../../../src/components/shared/Snack/Snack';
+
+[In our first part](https://www.notjust.dev/blog/2022-11-05-iOS16-lock-screen-ui), we started with the iOS 16 lock screen and we managed to build the awesome UI together. Let’s just review what we have built so far
+
+![Preview](./ui-preview.png)
+
+As you can see from the preview we built 3 main components in our screen,
+
+1. Header section → Where we show the lock icon and date-time.
+2. Notification list
+3. Footer → Where we show the flashlight and camera icon.
+
+As of now our date and time inside header are fixed which is really boring! Let’s change it. This is going to be the second part of the series where we would like to create a dynamic time component that will give us an updated date and time. Just like the clock on your wall.
+
+## Install `dayjs` package
+
+First we will do some prep work. In order to deal with date and time there are lots of libraries which help us make our life easier. In this project, we are going to use [dayjs](https://day.js.org/) library. Why dayjs?
+
+1. It has lots of utility functions for us. Formatting, validating, and parsing dates are really simple!
+2. The library size is really small.
+
+To install this library just run,
+
+```jsx
+npm install dayjs
+```
+
+## Render the date-time
+
+We can render the date and time the way we want it very easily by the help of dayjs library
+
+- **Import the library**
+
+```jsx
+import dayjs from 'dayjs';
+```
+
+- **Display current date**
+  - To display the date, we can use `dayjs().format("dddd, DD MMMM")` that will format the date like `Friday, 05 November`
+  - To display the time, we can use `dayjs().format("hh:mm")` that will format the time like `12:00`
+
+With these changes, the texts at the top of our component should be,
+
+```html
+<Text style="{styles.date}">{dayjs().format("dddd, DD MMMM")}</Text>
+<Text style="{styles.time}">{dayjs().format("hh:mm")}</Text>
+```
+
+Cool! Now we should be able to see the current date and time in our simulators or devices. However, the work is not done yet. As you have probably discovered that our date-time is not really updating, even though it gives us the current date-time when the application renders but after that it does not update anymore.
+
+Let’s now fix it. We don’t want a broken clock in our application. Do we?
+
+## Update the date-time
+
+We need a couple of changes now in order to update the time.
+
+### Create a date-time state
+
+1. Since we will need to update our time, we need to put it into a state variable.
+
+```jsx
+import { useState } from 'react';
+
+// inside our component
```

Comment:
In this case, I usually put the declaration of the component here. It will still be one line (with either a comment line, or the actual declaration), but I think it's more clear where they should put it

```
import { useState } from 'react';

const Comp = () => {
   const [date, setDate] = useState(daysjs())
   ....
}